### ISSUE TITLE
Release agda2hs 1.0

### ISSUE THREAD

**Issue by jespercockx** [agda/agda2hs#163]
This repository is finally getting to a state that is ready for release on Hackage. This issue is here to collect the tasks that need to be done to complete the release:

- [x] Update the [license](https://github.com/agda/agda2hs/blob/master/LICENSE) with the current author list
- [x] Double-check the information in the cabal file
- [x] Request package maintainer status on Hackage (I have already done this)
- [x] Make a tag on github for the release candidate
- [x] Upload a package candidate to Hackage
- [x] Ask people to test out the candidate
  - [x] Fix #142 
  - [x] Update cabal version number to 1.0
- [x] Make a tag on github for the final release
  - [x] Remove `-Werror` from cabal file
  - [x] Remove smuggler from cabal file
- [x] Make the final release

It's probably also a good idea to check if there are any steps from https://github.com/agda/agda/blob/master/notes/agda-releases.txt that we should adapt, and create a similar file to document the process for the next release.

**Comment by jespercockx** [comment]
Currently blocked on https://github.com/agda/agda2hs/issues/154

**Comment by omelkonian** [comment]
Here's a script I use to automatically publish a Haskell package on version changes via the Github CI:
https://github.com/omelkonian/hs-pattrans/blob/master/publish.sh

The hackage credentials can be safely set using Github secrets:
https://github.com/omelkonian/hs-pattrans/blob/master/.github/workflows/ci.yml#L77-L94

Of course, incorporating the candidates will require a bit of tweaking (e.g. putting 'rc' suffixes, etc...), as well as the proper git tags, changelogs, etc... so it might make sense not to automate the whole process but have one person be responsible for these (that'd be you I guess).

**Comment by jespercockx** [comment]
I've created a tag for the first release candidate: https://github.com/agda/agda2hs/tree/v0.1.20230328

**Comment by jespercockx** [comment]
I've uploaded a release candidate to Hackage: https://hackage.haskell.org/package/agda2hs-0.1.20230328/candidate. Please give it a try to make sure it works as expected! I'll also post this on Zulip.

**Comment by omelkonian** [comment]
Is is a problem that the backend version is 0.1 (the one you get from `agda2hs --version`)?

**Comment by jespercockx** [comment]
We should probably also still fix https://github.com/agda/agda2hs/issues/142 before the release.

**Comment by jespercockx** [comment]
Tag for the final release: https://github.com/agda/agda2hs/releases/tag/v1.0

**Comment by jespercockx** [comment]
`agda2hs` version 1.0 has now officially been released on Hackage at https://hackage.haskell.org/package/agda2hs-1.0. Thank you all @UlfNorell @omelkonian @sarajuhosova @jmchapman @flupe @Riib11 for your help to make this possible!
